### ISSUE TITLE
RFC - consider adding expire (date) to multi-edit tool

### ISSUE THREAD

**Issue by phiw13** [textpattern/textpattern#1585]
What the headline says, for the Articles panel. This would make batch processing of the expire date (add/remove) much easier.

For context, see [this forum post](https://forum.textpattern.com/viewtopic.php?id=51191).

TY.

**Comment by petecooper** [comment]
+1 from me.

Further considerations: given the process of expiring an article comprises setting an expire date of `NOW()`, I'm also interested in other-way batch processing in a similar fashion:

* un-expire an article (i.e. remove any expiry date)
* batch 'reset timestamp to now'

**Comment by Bloke** [comment]
Oooh, yes please. With all the options Pete says, too.

**Comment by petecooper** [comment]
With that enthusiasm, sold!

**Comment by Bloke** [comment]
The only snagola with this is how to present the sub-options. Traditionally, it's been a dropdown of selections but that won't work here because a "Change expiry" multi-edit feature requires:

1) A 'none' option (easy: the empty dropdown entry, or a textbox with nothing in it).
2) A 'now' option (easy: a dedicated dropdown entry, or a textbox that defaults to the time 'now' (ish) or more likely the time the page was last loaded).
3) A custom date/timestamp (Not so easy. Requires either a) a bunch of input fields like we have on the Write panel (ugh), b) a date picker (that can also permit 'empty' dates to satisfy the ability to clear the expiry), or c) a freeform textbox to type the date and time in.

I don't really want to add three new entries to the multi-edit tool (_change expiry to..._, _reset expiry to now_, and _remove expiry_) so I'd like some form of combined option. Lowest common denominator is a textbox with some input validation on it or textbox that has `type="datetime"` and hope that the browser handles it well.

In theory, a native browser date picker solves everything, as it allows an arbitrary date, a 'today' (now) button and a 'none' button. But the kicker is that it's not so easy to type the date in, which is often faster than faffing with swiping/scrolling through years, months, days, hours, minutes, etc. Of those browsers that do permit dates to be typed, they're prescriptive on format (I believe) and only permit YYYY-MM-DD format, and possibly YYYY-MM-DDTHH:MM:SS for datetimes. Not sure: I'd need to check.

So what's the best way forward in terms of UX here? Anyone got any thoughts?

**Comment by petecooper** [comment]
>So what's the best way forward in terms of UX here? Anyone got any thoughts?

Defer until 4.9, without wishing to derail @phiw13's OP.

The work planned for 4.9 would make this complementary. Gut feeling says this is outside the scope of a patch release at (potentially) the end of a release series.

**Comment by Bloke** [comment]
That may be prudent.

**Comment by phiw13** [comment]
> Defer until 4.9, without wishing to derail @phiw13's OP.

Yes, please. That was the reason why I waited and only filed this _after_ the TXP 4.8.4 release. :-)

That said, my first thought was – KISS – an option to add, and remove, an option to set that date to `now()`. If I am allowed to be greedy, I would of course want the whole stuff, but that present serious UI difficulties and the eventual coding complexities. On the Write panel, it is all simple (by some definition of simple…).
